From 21d5b651bbc4bcad3656a1804692c70e32797c69 Mon Sep 17 00:00:00 2001
From: David Lamparter <equinox@diac24.net>
Date: Thu, 2 Apr 2020 21:16:04 +0200
Subject: [PATCH] bgpd, ospfd, ospf6d: long is not bool :(

... Oops ...

(for context, the defaults code originally didn't have a dedicated
"bool" variant and just used long for bools...  I derp'd this when
adding bool as a separate case :( )

Reported-by: Donald Sharp <sharpd@cumulusnetworks.com>
Signed-off-by: David Lamparter <equinox@diac24.net>
(cherry picked from commit 4c1458b595282bff6a6e0b20767bb5cb655d0b4c)
---
 bgpd/bgp_vty.c     | 16 ++++++++--------
 ospf6d/ospf6_top.c |  4 ++--
 ospfd/ospf_vty.c   |  4 ++--
 3 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/bgpd/bgp_vty.c b/bgpd/bgp_vty.c
index 243822206c..4f5ba285aa 100644
--- a/bgpd/bgp_vty.c
+++ b/bgpd/bgp_vty.c
@@ -73,20 +73,20 @@
 #endif
 
 FRR_CFG_DEFAULT_BOOL(BGP_IMPORT_CHECK,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 FRR_CFG_DEFAULT_BOOL(BGP_SHOW_HOSTNAME,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 FRR_CFG_DEFAULT_BOOL(BGP_LOG_NEIGHBOR_CHANGES,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 FRR_CFG_DEFAULT_BOOL(BGP_DETERMINISTIC_MED,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 FRR_CFG_DEFAULT_ULONG(BGP_CONNECT_RETRY,
 	{ .val_ulong = 10, .match_profile = "datacenter", },
diff --git a/ospf6d/ospf6_top.c b/ospf6d/ospf6_top.c
index 95537eb86e..ba3c1b8907 100644
--- a/ospf6d/ospf6_top.c
+++ b/ospf6d/ospf6_top.c
@@ -52,8 +52,8 @@
 DEFINE_QOBJ_TYPE(ospf6)
 
 FRR_CFG_DEFAULT_BOOL(OSPF6_LOG_ADJACENCY_CHANGES,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 
 /* global ospf6d variable */
diff --git a/ospfd/ospf_vty.c b/ospfd/ospf_vty.c
index 152a7e83b7..92c9191801 100644
--- a/ospfd/ospf_vty.c
+++ b/ospfd/ospf_vty.c
@@ -54,8 +54,8 @@
 #include "ospfd/ospf_bfd.h"
 
 FRR_CFG_DEFAULT_BOOL(OSPF_LOG_ADJACENCY_CHANGES,
-	{ .val_long = true, .match_profile = "datacenter", },
-	{ .val_long = false },
+	{ .val_bool = true, .match_profile = "datacenter", },
+	{ .val_bool = false },
 )
 
 static const char *const ospf_network_type_str[] = {
